File: frontend/src/types/analysis.ts
// Types para análisis - Extender api.ts existente
export interface AnalysisCreate {
  lot_id: string
  resource_type: ResourceType
}

export type ResourceType = 'resource_1' | 'resource_2' | 'resource_3' | 'resource_4'

export interface AnalysisRead {
  id: string
  lot_id: string
  resource_type: ResourceType
  average_rate: number
  min_value?: number
  max_value?: number
  std_dev?: number
  percentiles?: Record<string, number>
  raster_file_reference: string
  analyzed_at: string
  status: 'pending' | 'processing' | 'completed' | 'failed'
  error_message?: string
}

export interface AnalysisListResponse {
  items: AnalysisRead[]
  total: number
  page: number
  size: number
}

export interface TimeSeriesPoint {
  analyzed_at: string
  average_rate: number
  resource_type: ResourceType
}

export interface TimeSeriesResponse {
  lot_id: string
  resource_type: ResourceType
  data: TimeSeriesPoint[]
}

File: backend/app/schemas/analysis.py
# Schemas de análisis - Pydantic
from typing import Optional, List, Dict, Any
from pydantic import BaseModel, Field
from datetime import datetime
from uuid import UUID


class ResourceTypeEnum(str):
    RESOURCE_1 = "resource_1"
    RESOURCE_2 = "resource_2"
    RESOURCE_3 = "resource_3"
    RESOURCE_4 = "resource_4"


class AnalysisCreate(BaseModel):
    lot_id: UUID
    resource_type: ResourceTypeEnum


class AnalysisRead(BaseModel):
    id: UUID
    lot_id: UUID
    resource_type: ResourceTypeEnum
    average_rate: float
    min_value: Optional[float] = None
    max_value: Optional[float] = None
    std_dev: Optional[float] = None
    percentiles: Optional[Dict[str, float]] = None
    raster_file_reference: str
    analyzed_at: datetime
    status: str  # pending, processing, completed, failed
    error_message: Optional[str] = None

    model_config = {"from_attributes": True}


class AnalysisListResponse(BaseModel):
    items: List[AnalysisRead]
    total: int
    page: int
    size: int


class TimeSeriesPoint(BaseModel):
    analyzed_at: datetime
    average_rate: float
    resource_type: ResourceTypeEnum


class TimeSeriesResponse(BaseModel):
    lot_id: UUID
    resource_type: ResourceTypeEnum
    data: List[TimeSeriesPoint]